### DISCUSSION TITLE
Update certificates - issue with proxy-init.

### DISCUSSION THREAD

**Discussion by aaronolds** [linkerd/linkerd2#8895] [General] [answered]
**What is the Question?**
I am trying to update the certificates on my K8 cluster on AWS.  And I am getting the following error:

there are nodes using the docker container runtime and proxy-init container must run as root user.
try installing linkerd via --set proxyInit.runAsRoot=true
error: no objects passed to apply

Is there any way I can change this on an existing installation?

**Comment by alpeb** [comment]
To update linkerd's config you can use the `linkerd upgrade` or `helm upgrade` (depending on your method) commands, even if you're not performing a version upgrade. The `--set proxyInit.runAsRoot=true` flag can be used in both methods.

**Reply by aaronolds** [reply]
I tried that, but I cannot run that command because my issue credentials are not valid
`failed to validate issuer credentials: not valid anymore. Expired on 2022-07-14T19:44:09Z`

And I cannot update the issue credentials because:
`there are nodes using the docker container runtime and proxy-init container must run as root user.
try installing linkerd via --set proxyInit.runAsRoot=true
error: no objects passed to apply`

So I am not sure what to do?

**Comment by alpeb** [comment, marked as answer]
Ah I see... In that case you can try regenerating your issuer certs as explained [here](https://linkerd.io/2.11/tasks/generate-certificates/) and pass them to the upgrade command along with the `--set proxyInit.runAsRoot=true` flag:
```
linkerd upgrade --identity-trust-anchors-file ca.crt   --identity-issuer-certificate-file issuer.crt   --identity-issuer-key-file issuer.key --set proxyInit.runAsRoot=true
```

**Reply by aaronolds** [reply]
That worked!  Thanks!
